### PR TITLE
Feature/angular material and main design

### LINKED ISSUE #17: Add Angular Material and create main application design

### PR DESCRIPTION
**Pull request by laska-sl**

### CODE REVIEW

File: ArcadiaParties-SPA/angular.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
               "src/assets"
             ],
             "styles": [
+              "./node_modules/@angular/material/prebuilt-themes/pink-bluegrey.css",
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/angular.json]: No, this should be added to `styles.css`. To be more accurate, `styles.css` should be transformed to `styles.scss` and import this theme.

File: ArcadiaParties-SPA/src/app/content/content.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* tslint:disable:no-unused-variable */
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/content/content.component.spec.ts]: We decided with @Duveshka that you don't have to spend time for tests in this specific project.

File: ArcadiaParties-SPA/src/app/header/header.component.css [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.example-icon {
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/header/header.component.css]: You should use scss files to declare styles. It allows to use much more features than raw css.

File: ArcadiaParties-SPA/package.json [hunk 4]
```diff
@@ -12,10 +12,12 @@
   "private": true,
   "dependencies": {
     "@angular/animations": "~9.0.1",
+    "@angular/cdk": "^9.0.1",
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/package.json:15]: Do you use it somewhere?

File: ArcadiaParties-SPA/src/app/app.component.html [hunk 5]
```diff
@@ -1,3 +1,2 @@
-<h1>WELCOME</h1>
-<h1> Title is: {{ title$ | async }} </h1>
- 
\ No newline at end of file
+<app-header [title]=" title$ | async "></app-header>
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.html]: As header will do much more than just show the page title, it makes sense to encapsulate all things about header inside it.

File: ArcadiaParties-SPA/src/styles.css [hunk 6]
```diff
@@ -1 +1,9 @@
-/* You can add global styles to this file, and also import other style files */
+html, body { height: 100%; }
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/styles.css]: Could you use code formatting also for styles?

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/header/header.component.scss [hunk 7]
```diff
@@ -0,0 +1,7 @@
+.example-icon {
+    padding: 0 14px;
+  }
+  
+  .example-spacer {
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/header/header.component.scss]: Please, apply code formatting here too.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/package.json:15]: It was automatically added after angular-material installation. Should I delete this line?

File: ArcadiaParties-SPA/angular.json (hunk 1 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/angular.json]: Material theme reference changed. Changed CSS to SCSS

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/styles.css]: Code formatted

File: ArcadiaParties-SPA/src/app/content/content.component.spec.ts (hunk 2 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/content/content.component.spec.ts]: spec files removed

File: ArcadiaParties-SPA/src/app/header/header.component.css (hunk 3 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/header/header.component.css]: Changed CSS to SCSS

File: ArcadiaParties-SPA/src/app/header/header.component.scss (hunk 7 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/header/header.component.scss]: Code formatted

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/styles.css]: Code formatted

File: ArcadiaParties-SPA/src/app/app.component.html (hunk 5 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.html]: Encapsulated prop. Now header takes data from state.

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/package.json:15]: How did you add angular-material? If you run `npm i @angular/material -S` it should not had added another packages.

File: ArcadiaParties-SPA/src/app/app.module.ts [hunk 8]
```diff
@@ -2,26 +2,34 @@ import { BrowserModule } from '@angular/platform-browser';
...
-      runtimeChecks: {
-        strictStateImmutability: true,
-        strictActionImmutability: true
-      }
-    }),
-    StoreDevtoolsModule.instrument({ maxAge: 25, logOnly: environment.production })
-  ],
-  providers: [],
-  bootstrap: [AppComponent]
+   declarations: [
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.module.ts]: This file had indent by 2 spaces and now it has 3 spaces.

File: ArcadiaParties-SPA/src/app/header/header.component.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Store, select } from '@ngrx/store';
+import { State } from '../reducers/reducers';
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/header/header.component.ts]: As I already told @Duveshka, it is code style convention to place external imports before internal imports and separate them with empty line.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 8 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: These changes were at first commit in this PR. That's mean only differences between @Duveshka's workspace and mine.

File: ArcadiaParties-SPA/src/app/header/header.component.ts (hunk 9 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/header/header.component.ts]: I've fixed this.

**Comment by laska-sl** [comment]
At this stage app looks like
image

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/styles.css]: No, it isn't. Formatted code should look like:
```scss
html, body {
  ...
}

body {
  ...
}
```

**Comment by laska-sl** [comment]
At this stage app looks like
image
image

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 8 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Yes, they were added in your commit. Correct indent is 2 spaces (or 4 spaces), but you somehow made it 3.

File: ArcadiaParties-SPA/src/styles.scss [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+html, body { 
+    height: 100%; 
+}
+
+body { margin: 0; 
+    font-family: Roboto, "Helvetica Neue", sans-serif;
+    background-color: #edebe9;
+    display: flex;
+    overflow: scroll;
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/styles.scss]: No. It should not be `body` style.

File: ArcadiaParties-SPA/src/app/app.component.html [hunk 11]
```diff
@@ -1,3 +1,2 @@
-<h1>WELCOME</h1>
-<h1> Title is: {{ title$ | async }} </h1>
- 
\ No newline at end of file
+<app-header></app-header>
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.html]: I meant it a bit different way.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 8 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Yes, they were added in your commit. Correct indent is 2 spaces (or 4 spaces), but you somehow made it 3.

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/styles.css]: No, it isn't. Formatted code should look like:
```scss
html, body {
  ...
}

body {
  ...
}
```

Comment (laska-sl) [reply ArcadiaParties-SPA/src/styles.css]: But It already looks like this...

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 8 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Ok, I'll use Prettier. Problem has gone.

Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Then add editor config file to specify project settings for Prettier. Something like this:
```
# Editor configuration, see https://editorconfig.org
root = true

[*]
charset = utf-8
max_line_length = 120
end_of_line = crlf
trim_trailing_whitespace = true
insert_final_newline = true
indent_style = space
indent_size = 2
quote_type = single
```

Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.module.ts]: And also add Prettier extension for VS Code and set default code formatter: https://github.com/prettier/prettier-vscode#using-command-palette-cmdctrl--shift--p

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/styles.css]: Now it looks like this:
```css
body { margin: 0; 
  ...
}
```

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/package.json:15]: I used `ng add @angular/material` like in official guide. 

> The ng add command will install Angular Material, the Component Dev Kit (CDK), Angular Animations

File: ArcadiaParties-SPA/src/app/app.component.html (hunk 11 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.html]: This way?
image

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/package.json:15]: Ok, now you can remove CDK dependency from the project, you won't need it at all.

File: ArcadiaParties-SPA/src/app/app.component.html (hunk 11 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.component.html]: Yes.

File: ArcadiaParties-SPA/src/app/app.component.scss [hunk 12]
```diff
@@ -0,0 +1,11 @@
+.header-tollbar {
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.scss]: Typo in `tollbar`.

File: ArcadiaParties-SPA/src/app/app.component.scss [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+  position: fixed;
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.scss]: Is not needed.

File: ArcadiaParties-SPA/src/app/app.component.scss [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+  top: 0;
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.scss]: Is not needed too.

File: ArcadiaParties-SPA/src/app/app.component.scss [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+  height: 56px;
+  z-index: 2;
+}
+
+.container {
+  position: absolute;
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.scss]: No :)

File: ArcadiaParties-SPA/src/app/app.module.ts [hunk 16]
```diff
@@ -1,27 +1,34 @@
-import { BrowserModule } from '@angular/platform-browser';
-import { NgModule } from '@angular/core';
-import { StoreModule } from '@ngrx/store';
-import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { BrowserModule } from "@angular/platform-browser";
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.module.ts]: Quotes should be single, as you specified it in editorconfig.

File: ArcadiaParties-SPA/src/app/app.module.ts [hunk 17]
```diff
@@ -1,27 +1,34 @@
...
+import { NgModule } from "@angular/core";
+import { StoreModule } from "@ngrx/store";
+import { StoreDevtoolsModule } from "@ngrx/store-devtools";
+import { MaterialModule } from "./material/material.module";
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.module.ts]: Internal after external.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/package.json:15]: Ok, after some investigation I realized that I was wrong and this project is needed for @angular/material correct work.

File: ArcadiaParties-SPA/src/app/header/header.component.html [hunk 18]
```diff
@@ -0,0 +1,4 @@
+<span>{{ title$ | async }}</span>
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/header/header.component.html]: When you use flex it doesn't matter whether to use divs or spans, but divs are more commonly used.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/package.json (hunk 4 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/package.json:15]: @angular/material and @angular/cdk returned.

File: ArcadiaParties-SPA/src/styles.css (hunk 6 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/styles.css]: Code formatted again

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 8 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Indent configured.

File: ArcadiaParties-SPA/src/styles.scss (hunk 10 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/styles.scss]: Style files changed.

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 12 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss]: Fixed typo.

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 13 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss]: Position changed.

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 14 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss]: And this position changed.

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 16 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Quotes fixed.

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 17 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: I've fixed this.

File: ArcadiaParties-SPA/src/app/header/header.component.html (hunk 18 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/header/header.component.html]: I've changed spans to divs

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 15 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss]: Now app.component has absolute position

Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.component.scss]: No :)

File: ArcadiaParties-SPA/src/app/app.module.ts [hunk 19]
```diff
@@ -2,26 +2,34 @@ import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
 import { StoreModule } from '@ngrx/store';
 import { StoreDevtoolsModule } from '@ngrx/store-devtools';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+import { MaterialModule } from './material/material.module';
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.module.ts]: Not needed empty line.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 19 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Line has deleted.

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 15 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss]: Style changed. No absolute positioning.

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 19 above)
Comment (sponkrashin) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Not the line that needed.

File: ArcadiaParties-SPA/src/app/content/content.component.scss [hunk 20]
```diff
@@ -0,0 +1,5 @@
+:host {
+  overflow: auto;
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/content/content.component.scss]: This is not content component's duty to set overflow. It is content container's responsibility.

File: ArcadiaParties-SPA/src/app/app.component.scss [hunk 21]
```diff
@@ -0,0 +1,11 @@
+.content-container {
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.scss:1]: It should also have `flex` and `display` styles.

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/app.component.scss (hunk 21 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.scss:1]: I've fixed this problem.

File: ArcadiaParties-SPA/src/app/content/content.component.scss (hunk 20 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/content/content.component.scss]: I've fixed this problem.

File: ArcadiaParties-SPA/src/app/app.module.ts (hunk 19 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.module.ts]: Empty line fixed.

File: ArcadiaParties-SPA/src/app/app.component.html [hunk 22]
```diff
@@ -1,3 +1,7 @@
-<h1>WELCOME</h1>
-<h1> Title is: {{ title$ | async }} </h1>
- 
\ No newline at end of file
+<mat-toolbar color="primary" class="header-container">
```
Comment (sponkrashin) [inline ArcadiaParties-SPA/src/app/app.component.html:1]: Does material library set style for toolbar by itself?

**Review by sponkrashin — CHANGES_REQUESTED** [review]

File: ArcadiaParties-SPA/src/app/app.component.html (hunk 22 above)
Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.html]: Yeah, but I've chosen this particular color style from the available.

Comment (laska-sl) [reply ArcadiaParties-SPA/src/app/app.component.html:1]: Yeah, but I chose this particular color style from the available. No need in this, we can do it later.

**Review by sponkrashin — APPROVED** [review]

**Pull request merged**